File: CSharp/Tests/ReverseTheCaseTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class ReverseTheCaseTest
    {
        [Theory]
        [InlineData("Happy Birthday", "hAPPY bIRTHDAY")]
        [InlineData("MANY THANKS", "many thanks")]
        [InlineData("sPoNtAnEoUs", "SpOnTaNeOuS")]
        [InlineData("eXCELLENT, yOuR mAJESTY", "Excellent, YoUr Majesty")]
        public void ReverseCase_StringInputValue_ReturnStringWithCharactersChangedCase(string str, string expected)
        {
            var actual = ReverseTheCase.ReverseCase(str);

            Assert.Equal(expected, actual);
        }
    }
}

File: js/tests/reverseTheCase.test.js
const reverseCase = require("../reverseTheCase");

const data = [
  ["Happy Birthday", "hAPPY bIRTHDAY"],
  ["MANY THANKS", "many thanks"],
  ["sPoNtAnEoUs", "SpOnTaNeOuS"],
  ["eXCELLENT, yOuR mAJESTY", "Excellent, YoUr Majesty"]
];

describe("reverseCase", () => {
  describe("Return string with characters in reverse case", () => {
    test.each(data)("reverseCase(%p) = %p", (str, result) => {
      let actual = reverseCase(str);

      expect(actual).toEqual(result);
    });
  });
});
